src/scenes/ObservationsRouter/details/index.tsx

```diff
@@ -108,6 +110,26 @@ export default function ObservationDetails(props: ObservationDetailsProps): JSX.
     return undefined;
   }, [activeLocale, details, observation]);
 
+  useEffect(() => {
+    const speciesWithNoIdNames = details?.species.filter((sp) => !sp.speciesId).map((sp) => sp.speciesName || '');
+    const speciesWithNoIdSet = new Set(speciesWithNoIdNames);
+    const speciesWithNoIdMap = Array.from(speciesWithNoIdSet);
```

Comment:
If this is here to dedupe the list, you might consider using `lodash`.

Something like this might work:
```
let speciesWithNoId = _.uniqBy((details?.species || []).filter((sp) => !sp.speciesId), 'speciesName')
```

Reply:
Thanks!